pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/consumer/PulsarKafkaConsumer.java

```diff
@@ -112,39 +113,54 @@ public PulsarKafkaConsumer(Map<String, Object> configs) {
         this(configs, null, null);
     }
 
-    public PulsarKafkaConsumer(Map<String, Object> configs, Deserializer<K> keyDeserializer,
-            Deserializer<V> valueDeserializer) {
-        this(new ConsumerConfig(ConsumerConfig.addDeserializerToConfig(configs, keyDeserializer, valueDeserializer)),
-                keyDeserializer, valueDeserializer);
+    public PulsarKafkaConsumer(Map<String, Object> configs, Schema<K> keyDeserializer,
+            Schema<V> valueDeserializer) {
+        this(configs, new Properties(), null, keyDeserializer, valueDeserializer);
     }
 
     public PulsarKafkaConsumer(Properties properties) {
         this(properties, null, null);
     }
 
-    public PulsarKafkaConsumer(Properties properties, Deserializer<K> keyDeserializer,
-            Deserializer<V> valueDeserializer) {
-        this(new ConsumerConfig(ConsumerConfig.addDeserializerToConfig(properties, keyDeserializer, valueDeserializer)),
-                keyDeserializer, valueDeserializer);
+    public PulsarKafkaConsumer(Properties properties, Schema<K> keyDeserializer,
+            Schema<V> valueDeserializer) {
+        this(new HashMap<>(), properties, null, keyDeserializer, valueDeserializer);
     }
 
     @SuppressWarnings("unchecked")
-    private PulsarKafkaConsumer(ConsumerConfig config, Deserializer<K> keyDeserializer,
-            Deserializer<V> valueDeserializer) {
+    private PulsarKafkaConsumer(ConsumerConfig consumerConfig, Schema<K> keyDeserializer,
+                                Schema<V> valueDeserializer) {
+        this(new HashMap<>(), new Properties(), consumerConfig, keyDeserializer, valueDeserializer);
+    }
+
+    @SuppressWarnings("unchecked")
+    private PulsarKafkaConsumer(
```

Comment:
why do we need `Map<String, Object> conf` and `Properties` here? Isn't `consumerConfig` is good enough?